### PR TITLE
StringBox + Numeric hidden from userland

### PR DESCRIPTION
**Pull request by samuelselleck**
This hides StringBox and Numeric (except for in nested structs) by introducing PaxValue, an enum containing variants for all common pax types (Color, Primitive integers, Floats, Fills, etc).

This comes with two traits:
- ToFromPaxValue makes inspecting and converting to/from interior types possible
- CoercingRules makes it possible to convert multiple interior types into the expected output type (including in expressions)

Numeric and StringBox still exist, but can hopefully be removed once Add/Sub etc is implemented for PaxValue and all Paxel expressions work with this type instead.

There's a //COMMENT-TO-REVIEWER in one spot where I don't feel very confident in the current solution, let me know if you have any ideas!

### CODE REVIEW

File: pax-manifest/src/deserializer/mod.rs [hunk 1]
```diff
@@ -34,76 +35,64 @@ impl Deserializer {
...
+                _ => Err(Error::UnsupportedMethod),
+            }
+        } else if let Ok(_ast) = PaxParser::parse(Rule::literal_number, str) {
+            Ok(from_pax::<Numeric>(str).unwrap())
+        } else {
+            Err(Error::UnsupportedMethod)
+        };
+        if let Ok(val) = ret {
+            return Ok(val);
+        }
```
Comment (samuelselleck) [inline pax-manifest/src/deserializer/mod.rs:95]: @warfaj This is the part I'm unsure about! Is there some way of integrating this into the deserializer in a better way?

Comment (samuelselleck) [reply pax-manifest/src/deserializer/mod.rs:95]: For context: These pre-checks were taken from the body of the IntoableLiteral parser.

**Review by warfaj — APPROVED** [review]
LGTM!

File: pax-manifest/src/deserializer/mod.rs (hunk 1 above)
Comment (warfaj) [reply pax-manifest/src/deserializer/mod.rs:95]: We can leave this for now and revisit when we turn PaxAny -> PaxValue.

**Comment by warfaj** [comment]
Looks like deserialize_enum is failing. Take a look and fix then you can merge

**Pull request merged**